### PR TITLE
refactor(server): auth dtos

### PR DESCRIPTION
**Pull request by jrasm91**
- Consolidate auth dtos
- Remove admin sign up response (in favor of user response dto)
- Deprecate `OAuthConfigResponseDto` (for the mobile devs)

Tested via:
- All e2e still pass
- Clean install + sign up works as expected

### CODE REVIEW

File: server/src/domain/auth/auth.dto.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+export class SignUpDto extends LoginCredentialDto {
+  @IsString()
+  @IsNotEmpty()
+  @ApiProperty({ example: 'Admin' })
+  firstName!: string;
+
+  @IsString()
+  @IsNotEmpty()
+  @ApiProperty({ example: 'Doe' })
+  lastName!: string;
```
Comment (danieldietzler) [inline server/src/domain/auth/auth.dto.ts:70]: We don't want to do #4841 in this PR I'm assuming?

Comment (jrasm91) [reply server/src/domain/auth/auth.dto.ts:70]: Correct.

File: server/src/domain/auth/auth.dto.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+export class ChangePasswordDto {
+  @IsString()
+  @IsNotEmpty()
+  @ApiProperty({ example: 'password' })
+  password!: string;
+
+  @IsString()
+  @IsNotEmpty()
+  @MinLength(8)
```
Comment (danieldietzler) [inline server/src/domain/auth/auth.dto.ts:81]: Do we want to have a more strict password policy?

**Review by danieldietzler — APPROVED** [review]

File: server/src/domain/auth/auth.dto.ts (hunk 2 above)
Comment (jrasm91) [reply server/src/domain/auth/auth.dto.ts:81]: I think it is ok for now

**Pull request merged**